src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -571,15 +572,95 @@ private unsafe int HandleEventPeerStreamStarted(ref PEER_STREAM_STARTED_DATA dat
     }
     private unsafe int HandleEventPeerCertificateReceived(ref PEER_CERTIFICATE_RECEIVED_DATA data)
     {
-        try
+        //
```

Comment:
The goal of QuicConnection.SslConnectionOptions was to separate certificate logic from the connection code. So that you can read this code like: "Ahh there is a cert validation logic, ok, let me get back to this if I need to". Now it's split over two places.
So please keep this event handler looking like it was.

Reply:
The validation logic is still inside `QuicConnection.SslConnectionOptions`. The part that I moved to `QuicConnection` deals with copying data from the `PEER_CERTIFICATE_RECEIVED_DATA` so that it can be handed off to the threadpool.